language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -129,49 +130,39 @@ main = do
         dependencies <-
           forM (Map.toList pm) $
             \(pkgId, (mbPkgName, pkg)) -> do
-                 let id = T.unpack $ unPackageId pkgId
-                     name = packageNameStr pkgId mbPkgName
+                 let id = unPackageId pkgId
+                     name = packageNameText pkgId mbPkgName
                      asName = if name == id then "itself" else name
-                 putStrLn $ "Generating " <> id <> " as " <> asName
+                 Data.Text.IO.putStrLn $ "Generating " <> id <> " as " <> asName
                  deps <- daml2ts opts pm pkgId pkg mbPkgName damlTypesVersion packageVersion
                  pure (name, name, deps)
         whenJust optInputPackageJson $
           writeTopLevelPackageJson optOutputDir dependencies
 
-packageNameStr :: PackageId -> Maybe PackageName -> String
-packageNameStr pkgId mbPkgIdent =
-  T.unpack (maybe (unPackageId pkgId) unPackageName mbPkgIdent)
+packageNameText :: PackageId -> Maybe PackageName -> T.Text
+packageNameText pkgId mbPkgIdent = maybe (unPackageId pkgId) unPackageName mbPkgIdent
 
 newtype Scope = Scope {unscope :: String}
 newtype Dependency = Dependency {undependency :: String}  deriving (Eq, Ord)
 
--- Gives the scope '@foo' given a directory path like '/path/to/foo'.
+-- Gives the scope 'foo' given a directory path like '/path/to/foo'.
```

Comment:
I found it easier to understand with the leading `@`. Here and below.

Reply:
I like it too but understand why the presence of `fromJust . stripPrefix` causes a maintainer to be concerned.

Reply:
That makes sense. Thanks for the explanation.

It could help if the comment explained very briefly that the scope `foo` is the `@foo` in a package name.

Reply:
Ok.

Reply:
This comment is going into https://github.com/digital-asset/daml/pull/4837. When `scope` becomes a command line argument, I'll be sure to move the comment along with it.

Reply:
This went back to the way we want it in https://github.com/digital-asset/daml/pull/4908 (making scope an optional command line argument).